### PR TITLE
Fix #76 McSwagger

### LINKED ISSUE #76: Swagger not working

### PR DESCRIPTION
**Pull request by Audakel**
Just go [here](http://localhost:8081/docs/api/view/index.html) when ur server is running.

### CODE REVIEW

**Review by kupiakos — COMMENTED** [review]
Can you move `SwaggerHandler` into its own class? Also, it should only implement `HttpHandler`, not `EndpointHandler`. It doesn't need `serverManager` it looks like either.

File: java/Phase 1.iml [hunk 1]
```diff
@@ -28,5 +28,6 @@
     <orderEntry type="library" name="byte-buddy-1.4.33" level="project" />
     <orderEntry type="library" name="objenesis-2.4" level="project" />
     <orderEntry type="library" name="annotations-13.0" level="project" />
+    <orderEntry type="library" name="org.apache.commons.io" level="project" />
```
Comment (kupiakos) [inline java/Phase 1.iml]: Also this doesn't need to be here.

**Comment by Audakel** [comment]
moved SwaggerHandler to own class, removed commons.io

File: java/src/server/client/EndpointHandler.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import server.games.IServerManager;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import static server.client.ServerCommunicator.sendResponse;
+
+public class EndpointHandler implements HttpHandler {
```
Comment (kupiakos) [inline java/src/server/client/EndpointHandler.java:12]: Nice, I was meaning to migrate this out.

File: java/src/shared/utils/FileUtils.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class FileUtils {
+
+    public static final Map<String, String> MIMETYPES = makeMimeTypes();
+
+    private static Map<String, String> makeMimeTypes() {
```
Comment (kupiakos) [inline java/src/shared/utils/FileUtils.java:11]: See [this SO answer](http://stackoverflow.com/a/847849/1530134) on how to normally do this. `makeMimeTypes` and `getMimeType` is not very clear, but whatevs

File: java/src/server/client/SwaggerHandler.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.logging.Logger;
+
+import static server.client.ServerCommunicator.sendResponse;
+
+/**
+ * Handles all the html and json requests to run swagger
+ * Appends ".json" to the request before getting the proper file from the file system
+ */
+public class SwaggerHandler implements HttpHandler {
+    private EndpointDispatcher getMethod, postMethod;
```
Comment (kupiakos) [inline java/src/server/client/SwaggerHandler.java:23]: I really don't think this needs to be here, or anything involving `EndpointDispatcher`.

**Pull request merged**